src/metabase/upload.clj

```diff
@@ -636,67 +640,74 @@
+          (driver/add-columns! driver
+                               (:id database)
+                               (table-identifier table)
+                               {auto-pk-column-keyword (driver/upload-type->database-type driver ::auto-incrementing-int-pk)}
+                               :primary-key [auto-pk-column-keyword]))
+
+        (scan-and-sync-table! database table)
+
+        (when create-auto-pk?
+          (let [auto-pk-field (table-id->auto-pk-column (:id table))]
+            (t2/update! :model/Field (:id auto-pk-field) {:display_name (:name auto-pk-field)})))
+
+        (events/publish-event! :event/upload-append
+                               {:user-id  (:id @api/*current-user*)
+                                :model-id (:id table)
+                                :model    :model/Table
+                                :details  {:db-id       (:id database)
+                                           :schema-name (:schema table)
+                                           :table-name  (:name table)
+                                           :stats       stats}})
 
-      (try
-        (driver/insert-into! driver (:id database) (table-identifier table) normed-header parsed-rows)
-        (catch Throwable e
-          (throw (ex-info (ex-message e) {:status-code 422}))))
-
-      (when create-auto-pk?
-        (driver/add-columns! driver
-                             (:id database)
-                             (table-identifier table)
-                             {auto-pk-column-keyword (driver/upload-type->database-type driver ::auto-incrementing-int-pk)}
-                             :primary-key [auto-pk-column-keyword]))
-
-      (scan-and-sync-table! database table)
-
-      (when create-auto-pk?
-        (let [auto-pk-field (table-id->auto-pk-column (:id table))]
-          (t2/update! :model/Field (:id auto-pk-field) {:display_name (:name auto-pk-field)})))
-
-      (events/publish-event! :event/upload-append
-                             {:user-id  (:id @api/*current-user*)
-                              :model-id (:id table)
-                              :model    :model/Table
-                              :details  {:db-id       (:id database)
-                                         :schema-name (:schema table)
-                                         :table-name  (:name table)
-                                         :stats       {:num-rows          row-count
-                                                       :num-columns       (count new-column-types)
-                                                       :generated-columns (if create-auto-pk? 1 0)
-                                                       :size-mb           (file-size-mb file)
-                                                       :upload-seconds    (since-ms timer)}}})
-
-      {:row-count row-count})))
+        (snowplow/track-event! ::snowplow/csv-append-successful api/*current-user-id* stats)
```

Comment:
We don't have a model in scope here - a model is created in `create-csv-upload!` when we first create the table, but the append action runs directly against the table. 

We could search for the first model created against this table, but that would be brittle as the model could have been archived, deleted, or edit in some semantic way.

Reply:
hmm, I think the model_id is desirable. I'd see a use case for it in which we want to know how of  X uploads were created, how many of them are then appended.

Reply:
I'm not fond of guessing the ID, I don't think there's enough value. I'm not sure how we query the snowcat cloud database, but imagine we can join on the upload event there and it'd be more robust.

Reply:
@qnkhuat 

> I'd see a use case for it in which we want to know how of X uploads were created, how many of them are then get appended.

We can calculate that statistic without the model_id, because there is a one-to-one relationship between models created from uploads and tables created from uploads.

We would know how many uploads were created: the number of model IDs from successful initial upload events.
We would know how many uploads have been appended to: the number of distinct table IDs from successful append events.